### PR TITLE
feat(bun/test): Impl. expect().pass() & expect().fail()

### PR DESCRIPTION
**Pull request by v3nntrnx**
expect().pass(), passing assertion: <https://jest-extended.jestcommunity.dev/docs/matchers/pass>
expect().fail(), failing assertion: <https://jest-extended.jestcommunity.dev/docs/matchers/fail>

### CODE REVIEW

File: src/bun.js/test/expect.zig [hunk 1]
```diff
@@ -289,6 +289,102 @@ pub const Expect = struct {
...
+            const value = arguments[0];
+            value.ensureStillAlive();
+
+            if (!value.isString()) {
+                globalObject.throwInvalidArgumentType("fail", "message", "string");
+                return .zero;
+            }
+
+            var _msg = value.toSliceOrNull(globalObject) orelse return .zero;
+            defer _msg.deinit();
```
Comment (Jarred-Sumner) [inline src/bun.js/test/expect.zig]: this is a use after free

in zig, `defer` is block scoped, not function scoped. in go, it is function scoped

Comment (v3nntrnx) [reply src/bun.js/test/expect.zig]: fixing in next commit, just wanted to say though that it worked

File: src/bun.js/test/expect.zig [hunk 2]
```diff
@@ -289,6 +289,100 @@ pub const Expect = struct {
...
+            const value = arguments[0];
+            value.ensureStillAlive();
+
+            if (!value.isString()) {
+                globalObject.throwInvalidArgumentType("pass", "message", "string");
+                return .zero;
+            }
+
+            var _msg = value.toSliceOrNull(globalObject) orelse return .zero;
+            defer _msg.deinit();
```
Comment (Jarred-Sumner) [inline src/bun.js/test/expect.zig]: UAF

Comment (v3nntrnx) [reply src/bun.js/test/expect.zig]: oh yeah, missed that. sorry

File: src/bun.js/test/expect.zig [hunk 3]
```diff
@@ -289,6 +289,98 @@ pub const Expect = struct {
...
+        if (arguments.len > 0) {
+            const value = arguments[0];
+            value.ensureStillAlive();
+
+            if (!value.isString()) {
+                globalObject.throwInvalidArgumentType("fail", "message", "string");
+                return .zero;
+            }
+
+            msg = value.toString(globalObject).toSlice(globalObject, default_allocator).slice();
```
Comment (Jarred-Sumner) [inline src/bun.js/test/expect.zig]: this leaks memory now

Comment (Jarred-Sumner) [reply src/bun.js/test/expect.zig]: make `msg` a `ZigString.Slice` and then use `value.toSlice(globalObject)` and it should be good

Comment (v3nntrnx) [reply src/bun.js/test/expect.zig]: On it

**Comment by Jarred-Sumner** [comment]
Thank you

**Comment by v3nntrnx** [comment]
> Thank you 

np <3

**Pull request merged**